src/cadenza.js

```diff
@@ -145,15 +145,21 @@ export class CadenzaClient {
    * @param {object} [options]
    * @param {boolean} [options.hideMainHeaderAndFooter] - Whether to hide the main Cadenza header and footer.
    * @param {boolean} [options.hideWorkbookToolBar] - Whether to hide the workbook toolbar.
+   * @param {string} [options.accept] - Media type; set to "application/pdf" for Jasper Report views
+   *     to show the PDF directly, without any Cadenza chrome around it.
    * @param {AbortSignal} [options.signal] - A signal to abort the iframe loading
    * @return {Promise<void>} A Promise for when the iframe is loaded
    * @throws For an invalid source
    */
-  show(source, { hideMainHeaderAndFooter, hideWorkbookToolBar, signal } = {}) {
+  show(
+    source,
+    { hideMainHeaderAndFooter, hideWorkbookToolBar, signal, accept } = {},
+  ) {
     this.#log('CadenzaClient#show', source);
     const params = new URLSearchParams({
```

Comment:
Proposal: Change the validMediaType() util to suppotedMediaType() and check the supported media types where applicable.

```
diff --git a/src/cadenza.js b/src/cadenza.js
index d777e23..346ae61 100644
--- a/src/cadenza.js
+++ b/src/cadenza.js
@@ -401,7 +401,7 @@ export class CadenzaClient {
    * Fetch data from a workbook view.
    *
    * @param {WorkbookViewSource} source - The workbook view to fetch data from
-   * @param {string} mediaType - The media type to use for the data
+   * @param {MediaType} mediaType - The media type to use for the data
    * @param {object} options - Options
    * @param {AbortSignal} [options.signal] - A signal to abort the data fetching
    * @return {Promise<Response>} A Promise for the fetch response
@@ -409,7 +409,10 @@ export class CadenzaClient {
    */
   fetchData(source, mediaType, { signal } = {}) {
     this.#log('CadenzaClient#fetchData', source, mediaType);
-    assert(validMediaType(mediaType), `Invalid media type: ${mediaType}`);
+    assert(
+      supportedMediaType(mediaType, [MediaType.CSV, MediaType.EXCEL]),
+      `Invalid media type: ${mediaType}`,
+    );
     const params = new URLSearchParams({ mediaType });
     return this.#fetch(resolvePath(source), { params, signal });
   }
@@ -442,14 +445,17 @@ export class CadenzaClient {
    * _Note:_ The file name, if not provided, is generated from the name of the workbook view and the current date.
    *
    * @param {WorkbookViewSource} source - The workbook view to download data from
-   * @param {string} mediaType - The media type to use for the data
+   * @param {MediaType} mediaType - The media type to use for the data
    * @param {object} options - Options
    * @param {string} [options.fileName] - The file name to use; The file extension is appended by Cadenza.
    * @throws For an invalid workbook view source or media type
    */
   downloadData(source, mediaType, { fileName }) {
     this.#log('CadenzaClient#downloadData', source, mediaType);
-    assert(validMediaType(mediaType), `Invalid media type: ${mediaType}`);
+    assert(
+      supportedMediaType(mediaType, [MediaType.CSV, MediaType.EXCEL]),
+      `Invalid media type: ${mediaType}`,
+    );
     const params = new URLSearchParams({
       mediaType,
       ...(fileName && { fileName }),
@@ -586,12 +592,23 @@ function validGeometryType(value) {
   ].includes(value);
 }
 
-/** @param {string} [value] */
-function validMediaType(value) {
-  const CSV = 'text/csv';
-  const MS_EXCEL_2007 =
-    'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet';
-  return value === CSV || value === MS_EXCEL_2007;
+/**
+ * @typedef {string} MediaType - A media type
+ * @see {@link https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types}
+ */
+
+const MediaType = /** @type {const Record<string, MediaType} */ {
+  CSV: 'text/csv',
+  EXCEL: 'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet',
+  PDF: 'application/pdf',
+};
+
+/**
+ * @param {MediaType} type
+ * @param {MediaType[]} supportedTypes
+ */
+function supportedMediaType(type, supportedTypes) {
+  return supportedTypes.includes(type);
 }
 
 /**
```

Reply:
Your patch does not allow PDF in downloadData(). I'll add that. Shall we also allow it in fetchData()? It is not explicitly required in the story, but I can't think of a reason for not allowing it.

Reply:
supportedMediaType() does not do much and is always used in combination with an assert. Shall we maybe put the assert insider the function and call it "assertMediaTypeSupported"?

Reply:
> Your patch does not allow PDF in downloadData(). I'll add that. Shall we also allow it in fetchData()? It is not explicitly required in the story, but I can't think of a reason for not allowing it.

Yes, please :+1: 

> supportedMediaType() does not do much and is always used in combination with an assert. Shall we maybe put the assert insider the function and call it "assertMediaTypeSupported"?

Sounds good :+1: I'd name it assertSupportedMediaType, though, so that the name is similar to the valid* functions.